### ISSUE TITLE
Cannot read property 'data' of undefined

### ISSUE THREAD

**Issue by ppo** [mattermost/mattermost#26803]
#### Summary
Error during account creation, via invitation.

#### Steps to reproduce
_(Sorry, the user can't explain me correctly what he did, he doesn't remember.)_

1. Send a member invitation.
2. Click on the link in the invitation email => opens Safari.
3. Follow the account creation process.

#### Expected behavior
An account creation process that doesn't crash due to a bug… 🤷‍♂️

#### Observed behavior (that appears unintentional) 

```
An unexpected error has occurred

Cannot read property 'data' of undefined

TypeError: Cannot read property 'data' of undefined at anonymous (/Users/runner/ Librarv/Developer/Xcode/DerivedData/ Mattermost-bkusffojapjudtaczuk-micpffbrx/Build/Intermediates.noindex/ Archivelntermediates/Mattermost/ BuildProductsPath/Release-inhoneos/ main.isbundle:734243:47)
at commitHookEffectListMount (/
Users/runner/Librarv/Developer/Xcode/ DerivedData/Mattermost-bkusffoigpiudtaczukmicpffbrx/Build/
Intermediates noindex/
Archivelntermediates/Mattermost/ BuildProductsPath/Release-iphoneos/ main isbundle:28072:371 at flushPassiveEffects I/Users/ runner/Librarv/Developer/Xcode/
DerivedData/Mattermost-bkusffoiapiudtaczukmicpffbrx/Build/
Intermediates noindex/
Archivelntermediates/Mattermost/ BuildProductsPath/Release-iphoneos/ main.isbundle:29177:46)
at anonvmous (/Users/runner/ Librarv/Developer/Xcode/DerivedDatal
Mattermost-bkusffoianiudtaczuk-[… truncated on the image as it's not me who had the problem]
```

#### Possible fixes
???

**Comment by amyblais** [comment]
@ppo What Mattermost server version are you on? What OS and version are you on?

**Comment by ppo** [comment]
I found this in my mobile app (and the user must have the same iPhone app version as mine)

App: 2.14.0 (Build 509)  
Server: 9.5.1 (Build 7877635852  
DB: postgres  
DB schema: 119

_(Hosted by [Framateam](https://framateam.org/))_

**Comment by amyblais** [comment]
@ppo It would be helpful if you're able to help share any of this information:
1. What Safari version are they on?
2. Do you see errors in the Mattermost server logs from the time the crash occurred?
3. At what point of the account creation process did they get the crash?

**Comment by ppo** [comment]
1. Safari in iOS 17.4.1
2. I don't have access to the server logs, it's a managed hosting (by Framateam).
3. As said earlier, the user can't explain me when exactly it happened. As the error message contains paths to Xcode/Mattermost, I guess after the redirection from Safari to the app (or auto-install of the app?). And as he doesn't have his password, before asking for that? (I created my account on my computer, not phone)

**Comment by amyblais** [comment]
Thank you, I haven't been able to reproduce this. Would your team be open to testing the account creation flow on our community server here https://community.mattermost.com/ to see if the same issue occurs there?

**Comment by amks1** [comment]
Someone from my team also got this error on Android. He's able to log in on the web though. 

Screenshot:
![image](https://github.com/mattermost/mattermost/assets/68153785/7847942c-0e95-4a0a-8d00-bc0f7332dd2b)

**Comment by amyblais** [comment]
Opened for investigation: https://mattermost.atlassian.net/browse/MM-58467

**Comment by lindy65** [comment]
Thanks for opening the issue @amyblais 

@ppo and @amks1 - please try updating your mobile apps to the latest available [here](https://mattermost.com/download/) - scroll down to mobile apps

I've been trying to reproduce the issue on Android on the latest build and am not able to.  I will ask a colleague to try to reproduce on iOS

**Comment by lindy65** [comment]
A question @ppo and @amks1 

Are the users creating their accounts in Mattermost on browser?  To ensure the best outcome, accounts should be created from a browser and then, once the user has their password, they can go to the mobile app and log in.

**Comment by ppo** [comment]
@lindy65 As I said, the user tapped on an invitation link received by email on his iPhone. Apparently it was opened in Safari. The Mattermost iOS app was not yet installed.

Now it's user account is created so upgrading the app won't be useful.

**Comment by lindy65** [comment]
Thanks @ppo 

You mention above that the user does not have his password?  Did he create his account in a browser?  If he created his account in a browser, he should have created a password at the same time?

So the error he received and that you posted in your first message was received in the browser?

**Comment by ppo** [comment]
@lindy65 I already gave here all the information I had. The user was not able to give me more information and now he has surely forgot.
As said, it crashed during the account creation process, I guess before he was asked for a password.
I don't know if he received the error in the browser… but based on the paths in the error log, it looks more like coming from an app, no?

**Comment by lindy65** [comment]
Thank you @ppo 

I think I have worked out the repro steps of your user and can reproduce his error on _first time clicking_ on "View in App" from the invite email after clicking "Join now" button

To me it seems the user was trying to create his account from his mobile (after clicking on the "Join now" button in the invite email

Once a user clicks on the "Join now" button in the email invite, their mobile browser opens and they are presented with a screen that says 
View in App / View in Browser

I think the user clicked on "View in App" as I clicked on this option and got the same error your user did.  However, after I clicked on "Relaunch", I was presented with the account creation screen "Let's get started".

Note though that once you complete the sign up / account creation via the browser on your mobile phone, you are logged into the mobile web version of Mattermost.  Not the actual Mattermost mobile app.

To log into the actual app, download it, add a server to log into and then log in (after having created your account and password from your browser)
